src/shared/components/copyDownloadControls/CopyDownloadQueryLinks.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {If} from 'react-if';
+import {ICopyDownloadInputsProps} from "./ICopyDownloadControls";
+
+export interface ICopyDownloadLinksProps extends ICopyDownloadInputsProps {
+    copyLinkRef?: (el: HTMLAnchorElement|null) => void;
+    showQuery?: boolean;
+    showVisualStudy?: boolean;
+    queryButtonLabel?: string;
+    visualStudyButtonLabel?: string;
+    handleQuery?: () => void;
+    handleVisualStudy?: () => void;
+}
+
+export class CopyDownloadQueryLinks extends React.Component<ICopyDownloadLinksProps, {}>
+{
+    public static defaultProps = {
+        className: "",
+        copyButtonLabel: "Copy",
+        downloadButtonLabel: "Download",
+        queryButtonLabel: "Query",
+        visualStudyButtonLabel: "Visual Study",
+        showCopy: true,
+        showDownload: true,
+        showQuery: false,
+        showVisualStudy: false
+    };
+
+    public render() {
+        return (
+            <span className={this.props.className}>
+                {
+                    this.props.showCopy &&
+                    <a onClick={this.props.handleCopy} ref={this.props.copyLinkRef}>
+                        <i className='fa fa-clipboard' style={{marginRight: 5}}/>{this.props.copyLabel}
+                    </a>
+                }
+                {
+                    this.props.showCopy && this.props.showDownload &&
+                    <span style={{margin: '0px 10px'}}>|</span>
```

Comment:
not a big deal but would be nicer to do this with css.   you could put a right border on all but the last <a> element in the list

Reply:
@alisman Thanks for your suggestion! I will try to address them all.